packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java

```diff
@@ -41,7 +42,22 @@ String resizeImageIfNeeded(
     try {
       String[] pathParts = imagePath.split("/");
       String imageName = pathParts[pathParts.length - 1];
-      File file = resizedImage(bmp, maxWidth, maxHeight, imageQuality, imageName);
+      SizeFCompat targetSize =
+          calculateTargetSize(
+              (double) originalSize.getWidth(),
+              (double) originalSize.getHeight(),
+              maxWidth,
+              maxHeight);
+      BitmapFactory.Options options = new BitmapFactory.Options();
+      options.inSampleSize =
+          calculateSampleSize(options, (int) targetSize.getWidth(), (int) targetSize.getHeight());
+      File file =
+          resizedImage(
+              decodeFile(imagePath, options),
```

Comment:
This can return null; it should still be read to a local variable and null checked, returning the original path if it's null as before. We shouldn't be depending on this returning null in exactly the same cases where `readFileDimensions` returns -`, especially since the implementation of that function could change later.